### PR TITLE
[TFL FE] Add STABLEHLO_COMPOSITE odml.rms_norm translation

### PR DESCRIPTION
**Pull request by akodanka**
### Details:
 - Add STABLEHLO_COMPOSITE odml.rms_norm translation
 - Reference for stablehlo.composite : https://openxla.org/stablehlo/spec#composite
 - Reference for `odml.rms_norm` (RMS Normalization) : https://github.com/google-ai-edge/litert-torch/blob/main/litert_torch/generative/layers/normalization.py#L193
 - This op is widely used in genai tflite/litert models like gemma3, gemma3n, gemma4.

### Tickets:
 - CVS-182852

### AI Assistance:
 - *AI assistance used: yes*
  - *Copilot was used to generate test code, and to fix errors in code*
  - *Manually reviewed the changes, and verified the test cases by running the tests*

### CODE REVIEW

File: src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    FRONT_END_GENERAL_CHECK(inputs.size() == 2,
+                            "STABLEHLO_COMPOSITE odml.rms_norm expects 2 inputs (data, gamma), got ",
+                            inputs.size());
+
+    const auto& data = inputs[0];
+    const auto& gamma = inputs[1];
+
+    double epsilon = static_cast<double>(node.get_attribute<float>("epsilon", 1e-6f));
+
+    auto rms = std::make_shared<ov::op::internal::RMS>(data, gamma, epsilon);
```
Comment (mvafin) [inline src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp:34]: Frontend should not create internal operations. Such operations are not serializable. Decompose Rope in a similar way pytorch frontend does it.

File: src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    // Dispatch to appropriate translator based on composite name
+    if (composite_name == "odml.rms_norm") {
+        return translate_odml_rms_norm(node);
+    }
+
+    // Unsupported composite operation
+    FRONT_END_OP_CONVERSION_CHECK(false,
+                                  "Unsupported STABLEHLO_COMPOSITE operation: ",
+                                  composite_name,
+                                  ". Currently only odml.rms_norm is supported.");
```
Comment (mvafin) [inline src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp:55]: `STABLEHLO_COMPOSITE` should also contain the decomposition for the operation stored inside. We should support converting such decompositions

**Review by mvafin — CHANGES_REQUESTED** [review]
Can we just support decompositions path for `STABLEHLO_COMPOSITE`? Will it be sufficient to fulfill the task if RMSNorm decomposed will fuse on compile time?

File: src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp (hunk 2 above)
Comment (akodanka) [reply src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp:55]: The idea behind the STABLEHLO_COMPOSITE op is to have a High-Level-Operation for a sequence of smaller mathematical operations, which can have an optimized hardware-specific implementation.
The decomposition is provided as a fallback mechanism.
Currently, we are already using the fallback mechanism in LiteRT, but that defeats the purpose of the composite op.
Minor changes in the decomposed graph can cause the mismatch of these patterns in the compiler, and the ops wouldn't get fused correctly.

Comment (taste1981) [reply src/frontends/tensorflow_lite/src/op/stablehlo_composite.cpp:55]: I second @akodanka on this. I can provide a few real-life examples we have with ISVs, on why we prefer STABLEHLO_COMPOSITE not to be decomposed: 

1.  For RMSNorm behind a STABLEHLO_COMPOSITE op, we see gamma might be either scalar of vector type. Current NPU RMS fuse pass does not handle vector type, so the fuse process ignores the gamma and it is very hard to notice this issue.  While if we take care of this in FE, all downstream plugin will benefit from this, instead of, for example, when NPU compiler fixes this, issue for GPU still persists.

2.. Consider a sliding window attention (SWA) that is wrapped into STABLEHLO_COMPOSITE. We see cases that ISV models split the KV cache between current token and history tokens, and concat the attention scores later. Fuse such kind of SWA variance into SDPA is actually not suitable to be added as part of NPU compiler pass or GPU common transformations, because it is model vendor specific.

My point is that if we don't prefer such kind of StableHLO to be directly mapped to RMS or SDPA, you'll need to add more and more customized patterns for them in the GPU/NPU plugin, and when you compile other models, they will also suffer from extended extra compilation time due to pattern match (though mostly mismatching these customer patterns).  This is just not scalable.

WDYT?

**Comment by akodanka** [comment]
Raised PR https://github.com/openvinotoolkit/openvino/pull/36263 to use ov::decomposition::rms_norm instead of ov::op::internal::RMS.
